DOCUMENTATION.md

```diff
@@ -107,19 +119,138 @@ In case it's a **dictionary** with fields, please make sure the **encoder** func
+
+### device
+Gives back the [**Device**](#device) object of the given id.
+```python
+client.device(dev_id)
+```
+- `dev_id`: **string**  the id of the device which is given back by the method.
+
+### devices
+Gives back the list of all the devices registered to the application.
+```python
+client.devices()
+```
+
+### update_device
+Updates an already existing device of the application.
+```python
+client.update_device(dev_id, updates)
+```
+- `dev_id`: **string**  the id of the device to be updated.
+- `updates`: **dictionary**  a dictionary with the fields to be updated in the device.
+
+### delete_device
+Deletes the device with the given id.
+```python
+client.delete_device(dev_id)
+```
+- `dev_id`: **string**  the id of the device to be deleted.
+
+### Device
+This objet is returned by the method `device()` of the ApplicationClient class. Here are its attributes:
+* `app_id`: **string**
+* `dev_id`: **string**
+* `latitude`: **float**
+* `longitude`: **float**
+* `altitude`: **float**
+* `description`: **string**
+* `attributes`: **dictionary**
+* `lorawan_device`: **dictionary**
+    * `app_eui`: **string**  8 bytes in hexadecimal
+    * `dev_eui`: **string**  8 bytes in hexadecimal
+    * `dev_addr`: **string**  4 bytes in hexadecimal
+    * `nwk_s_key`: **string**  16 bytes in hexadecimal
+    * `app_s_key`: **string**  16 bytes in hexadecimal
+    * `app_key`: **string**  16 bytes in hexadecimal
+    * `f_cnt_up`: **int**
+    * `f_cnt_down`: **int**
+    * `disable_f_cnt_check`: **boolean**
+    * `uses32_bit_f_cnt`: **boolean**
+
+### Application
+This object is returned by the method `get()` of the ApplicationClient class. Here are its attributes:
+* `app_id`: **string**
+* `payload_format`: **string**
+* `decoder`: **string**
+* `encoder`: **string**
+* `converter`: **string**
+* `validator`: **string**
+* `register_on_join_access_key`: **string**
+
+
+## Errors
+Errors can happen on connection or on some ApplicationClient's methods call, for different reasons:
 * Wrong `app_id`, `access_key` or `mqtt_address` were provided to the constructor.
-* The machine may not have access to the network/The MQTT server could be down/Firewall restrictions could prevent connection
+* The machine may not have access to the network/The MQTT server could be down/Firewall restrictions could prevent connection.
 * The client process doesn't have system capabilities to open a socket
-* The MQTT server uses MQTTS, but the client won't accept the TLS certificate
+* The MQTT server uses MQTTS, but the client won't accept the TLS certificate.
+* The Application client is not able to get the application or a device.
 Errors could also happen when closing connection, in case the disconnection is unexpected.
 It's possible to catch those exceptions using `except RuntimeError as` and print the error.
```

Comment:
In hindsight, I'm not sure we have to teach users how to catch exceptions. Perhaps rename the whole section "Exceptions", and remove this line?

Reply:
Also, make it clear that the mentioned error origins are only the most common errors, and that there can be other edge cases